### DISCUSSION TITLE
how to fix third-party laravel macro not detecting problem?

### DISCUSSION THREAD

**Discussion by ariaieboy** [larastan/larastan#1737] [Q&A]
Hi, I have a package that adds 2 macros to `Filament\Tables\Columns\TextColumn` [source](https://github.com/ariaieboy/filament-currency/blob/60183d5e8cc9cccf69675ac1c5d478ede025cdee/src/FilamentCurrencyServiceProvider.php#L27)
The phpstorm detects these macros without any problem, but when I use Larastan, I get these errors in my applications :
`Call to an undefined method Filament\Forms\Components\TextInput::currencyMask()` 

Is there any thing that I can do to fix this problem?

**Comment by canvural** [comment]
Does this happen in another application that uses your package, or happens at the package itself?

In any case, make sure that service provider runs when Larastan boots it. Usually that should be the case, and then Larastan can recognize the macro.

**Reply by ariaieboy** [reply]
It happens in another application that uses my package
